gusto/time_discretisation.py

```diff
@@ -470,16 +474,22 @@ def solve_stage(self, x_in, stage):
             self.x1.assign(x_in + 0.5 * self.dt * self.k1)
 
         elif stage == 1:
+            for evaluate in self.evaluate_source:
+                evaluate(x_in, self.dt)
```

Comment:
Should the source term be evaluated at `x1` instead of `xin`?
If I've read it correctly , `xin` is the solution at the beginning of the subcycle and doesn't get updated at each stage, whereas `x1` is the solution at each stage (ie the argument to the function F in the class doctring)

Reply:
Yes I think so! Thanks.